packages/polymath-issuer/src/index.js

```diff
@@ -1,48 +1,101 @@
-
-import Root from './components/Root';
-import store, { history } from './redux/store';
-import { unregister } from './registerServiceWorker';
-
-import { SENTRY_KEY } from './constants';
-import routes from './routes';
-
-unregister();
-
-if (process.env.NODE_ENV === 'production') {
-  // Init error monitoring tool
-  Sentry.init({
-    dsn: SENTRY_KEY,
-    environment: process.env.REACT_APP_DEPLOYMENT_STAGE,
+import { find } from 'lodash';
+/**
+ * This code is here to enable the progressive migration from Legacy to Modern.
+ * There are several levels in which Modern can replace Legacy functionality
+ * for a given route:
+ * 1. App level: Runs Modern on its own, Legacy is not used for this route
+ * 2. Router level: A subtree of Legacy is replaced by a subtree imported
+ * from Modern.
+ *
+ * There are two main capabilities that this setup allows:
+ * 1. To render the Modern or Legacy dApp depending on the current route.
+ * 2. To transform routes before mounting either app.
+ *
+ * NOTE: In theory we could also have logic to resolve values required for the
+ * redirected route that might not exist yet. Though this would require, an
+ * extra initialization step that might complicate things, so if possible lets
+ * avoid it.
+ */
+
+const redirectors = [
+  /*
+  Example:
+
+  {
+    pattern: /dashboard\/(.+)\/dividends/,
+    makeRoute: (params: string[]) => {
+      const [securityTokenSymbol] = params;
+      return `/securityTokens/${securityTokenSymbol}/dividends`;
+    },
+  },
+  
+  */
+];
+
+/**
+ * This is a map that determines wether the Legacy or the Modern App
+ * should be mounted. If there's no match, we default to Legacy.
+ */
+const modernAppRoutePatterns = [
+  // Example: /securityTokens\/(.+)\/dividends/
+];
+
+function getRoutePath() {
+  return window.location.href
+    .split('/')
+    .slice(3)
+    .join('/');
+}
+
+function initializeApp() {
+  // See if there is a redirection match and get the redirection route
+  // if so.
+  const path = getRoutePath();
+
+  let redirectionMatch;
+  let matchingRedirector;
+  redirectors.forEach(redirector => {
+    if (redirectionMatch) {
+      return;
+    }
+
+    redirectionMatch = path.match(redirector.pattern);
+    matchingRedirector = redirector;
   });
+
+  // If there's no redirection we render the proper app
+  if (!redirectionMatch) {
+    mountApp();
+    return;
+  }
+
+  // Parse parameters from redirector's regex
+  const params = redirectionMatch.slice(1);
+  const getTargetRoute = matchingRedirector.makeRoute(params);
+
+  // Redirect to route defined by redirector
+  window.location = getTargetRoute;
+}
+
+// Mounts the right app depending on the current route
+function mountApp() {
+  const path = getRoutePath();
+  let match;
+  modernAppRoutePatterns.forEach(pattern => {
+    if (match) {
+      return;
+    }
+    match = path.match(pattern);
+  });
+
+  if (match) {
+    console.log('Mounting: Modern dApp');
+    require('./mountModern');
```

Comment:
Maybe you could use Webpack code-splitting instead here:
`import('./mountModern')`
This way when you load the app the first time, you would actually load only 1 app. That might not change much because we have zero performances optimisation but as the current bundle is 8mb, you would load 8mb * 2 of JS bundle if you load the 2 apps, might start to be a little big 😄

Reply:
Yeah we could do that. I figured it was not worth the effort given this is only going to live for ~2 months

Reply:
If this is a quick win we should do it though. TBH I've never had to use dynamic imports before, you think this would be easy?

Reply:
Well, let's tackle this in a separate PR if we want to so that we don't postpone this merge any further. I know @josepmc is impatient